src/k8s/pkg/k8sd/app/hooks_bootstrap.go

```diff
@@ -348,18 +347,11 @@ func (a *App) onBootstrapControlPlane(s *state.State, bootstrapConfig apiv1.Boot
 		}
 	}
 
-	cmData, err := cfg.Kubelet.ToConfigMap()
-	if err != nil {
-		return fmt.Errorf("failed to format kubelet configmap data: %w", err)
-	}
-
-	client, err := k8s.NewClient(snap.KubernetesRESTClientGetter(""))
-	if err != nil {
-		return fmt.Errorf("failed to create kubernetes client: %w", err)
-	}
-
-	if _, err := client.UpdateConfigMap(s.Context, "kube-system", "k8sd-config", cmData); err != nil {
-		return fmt.Errorf("failed to update node configs: %w", err)
+	// Trigger an update of the configuration.
+	// Do not wait if the channel is full. The reconcilation loop will apply the most recent changes
+	select {
```

Comment:
This seems alright for now although if we ever depend on that config to be present after this section to do more  bootstrap stuff we might introduce a race condition or some weird errors here. Nothing to worry about, just some food for thought.

Reply:
There should be no problem. There are two scenarios:
a) the config needs to be set on bootstrap and cannot be changed later.
-> then the config should be applied before this part already and not get the config from the config map anyway.
b) the config can be changed on runtime. Then the reconciliation loop will apply the changes eventually and restart the relevant services. The services might not work in the time between config update and restart but this is fine.